doc/tutorials/highlevel.md

```diff
@@ -0,0 +1,322 @@
+```
+
+Please note that, while it shouldn't happen, if you setup everything correctly, calling a getter on a non-existent, wrongly typed or otherwise
+inconvertible key is a fatal error. All fatal errors result in a call to the fatal error handler and therefore exit the application.
+
+### Writing config values
+
+Writing config values is just as easy as reading them.
+
+```c
+ElektraError * error = NULL;
+ELEKTRA_SET (ELEKTRA_TAG_MYDOUBLE) (elektra, 3.141593, &error);
+if (error == NULL) {
+    // handle error
+    elektraErrorReset (&error);
+}
+```
+
+Admittedly writing is not quite as easy as reading, because of the error handling, but that is just a necessary evil when working in C. It
+is important to call `elektraErrorReset`, if an error was returned. Calling a setter with a non-null error pointer is a fatal error.
+
+Of course you can also use `elektraSet` (error handling omitted):
+
+```c
+elektraSet (elektra, ELEKTRA_TAG_MYDOUBLE, 3.141593, &error);
+elektraSetV (elektra, ELEKTRA_TAG_MYFLOATARRAY, 2.718282f, &error, 2);
+```
+
+Note that `elektraSetV` takes the `ElektraError` argument before the variable arguments, while in `ELEKTA_SET` the error is always the last
+argument. This is simply because of limitations in the C macro system.
+
+There is not setter for array sizes. Since Elektra's low-level part supports discontinuous arrays, we simply change the array size whenever
+necessary, if an array element setter is called. However, the high-level API has no support for discontinuous arrays, so take care not to
+create holes in your arrays, if you want to iterate over them. Remember, accessing non-existent keys (and this includes array elements) is a
+fatal error.
+
+### Advanced concepts
+
+The code-generator has some more advanced features that are supported out of the box. For example, you can read multiple config values at
+once by utilising structs. The use of structs also allows for recursive configurations like menus (a menu can have submenus).
+
+For more information take a look at the man-page [`kdb-gen-highlevel(1)`](../help/kdb-gen-highlevel.md).
+
+## Compiling your application
+
+Once you've written your application, you will want to compile it. This requires linking some libraries and adding to your include path.
+The easiest way is to use CMake or pkg-config to find the needed compiler options. Examples on how set this up can be found in
+[here](/examples/codegen/econf) and [here](/examples/codegen/tree).
+
+The compiler invocation should look something like this:
+
+```sh
+cc myapp.c conf.c `pkg-config --cflags --libs elektra-codegen` -I. -o myapp -Wl,-rpath `pkg-config --variable=libdir elektra-codegen`
+```
+
+Note: At least C99 is required, so if your compiler defaults to an older version you'll need to add `-std=c99`.
+
+## Running your application
+
+Running your application is easy, just run the executable (e.g. `myapp`). While this might work out of the box, you will just get the default
```

Comment:
For the application written in this guide it should work. It will not work, if there are no embedded defaults or if validation of the mountpoint is enabled and the mountpoint isn't there.

I will look into your error.

Reply:
I tried to stick to what is in the tutorial. In any case, you should clarify when it works, the "might" is very weak there.